File: trace-agent-bootstrap/src/main/java/com/vmlens/trace/agent/bootstrap/lock/LockExit.java
package com.vmlens.trace.agent.bootstrap.lock;

import com.vmlens.trace.agent.bootstrap.event.runtimeeventimpl.LockExitEvent;

public class LockExit implements LockOperation {

    @Override
    public LockEvent create(LockType lockType, Object object) {
        return new LockExitEvent(lockType,object);
    }

}

File: sync-bug/src/main/scala/com/vmlens/preanalyzed/model/lockoperation/LockExit.scala
package com.vmlens.preanalyzed.model.lockoperation

import com.vmlens.preanalyzed.model.{LockType, ReadLock, WriteLock, ReentrantLock}
import com.vmlens.trace.agent.bootstrap.preanalyzed.model.methodtypeimpl.AbstractMethodType
import com.vmlens.trace.agent.bootstrap.preanalyzed.model.methodtypeimpl
class LockExit extends LockOperation {

  override def create(lockType: LockType): AbstractMethodType = {
     lockType match {
       case ReadLock() => {
        methodtypeimpl.LockMethod.EXIT_READ_LOCK;
      }
      case WriteLock() => {
        methodtypeimpl.LockMethod.EXIT_WRITE_LOCK;
      }
      case ReentrantLock() => {
        methodtypeimpl.LockMethod.EXIT_REENTRANT_LOCK;
      }
    }
  }
}
